### PR TITLE
Start first tab for the App with configuration tab

### PR DESCRIPTION
**Pull request by f-hafner**
Creates functionality for running the app with 1 tab for the configuration.

### Main changes
- function `launch_application` to run the app. I largely followed examples on the internet. The function first loads the reference data, and then calls `inst/application/app.R`. The reference data are passed through shinyOptions. The default argument to `launch_application` is such that I can run it without arguments. We can change this later, but if you want it changed now, let me now.
- The app uses modules defined in `R/`. I created a module for both the simulation (`simulation_config_module.R` and for the program configuration (`program_config_module.R`). Each module has a `server` and `ui` component that are used in the app. The server component is tested with `testServer`; the UI component is not tested for now.
  - the `program_config_module` uses itself another module: `directory_config_module`. To test the `program_config_server`, I had to use some mocking
- Both modules return reactive values that will be used later by the app to call the java app `dynamo-hia`. I think the reactivity and return values are implemented correctly, but it's all new to me. We will see :) 
- Update README with instructions for running the app, naming conventions I adopted for modules, and decisions about testing

### Other changes
- minor changes to the old app so that it can still be called. useful for comparing UIs new/old without switching branches.

### What I'll still fix
- [ ] ~~refactor the function to test the elements of two lists~~ -> only used once, for now leave as is
- [x] consistently call shiny functions without the shiny:: prefix
- [x]  squash commits

### What I did not do and we can put to a wish list (new issue): experimenting with nicer configuration UI
- rearrange buttons in the simulation config
- experiment with nicer layouts for the buttons in the directory config

### CODE REVIEW

**Comment by f-hafner** [comment]
there is a test failing in the CI, but it works locally. will investigate

**Comment by f-hafner** [comment]
To change
- no argument for launch_application; let the user choose the working directory
- the simulation directory is not necessary; assume it's in the working directory under Simulation/
- for dyanmo, set a path to a file (the executable) instead a directory
- for shiny functions, don't use `shiny::`, but for functions from other packages, do

**Comment by f-hafner** [comment]
I updated the code according to Malte's suggestions. Compared the initial comments, the following has changed 
- the argument for the working directory in the `launch_application` function is removed. instead, when the user chooses the working directory, the app calls `get_reference_data`. If the data has the expected format (a very rough check only at the moment), it updates the `Population` field in the app. If it does not have the expected format, an error message is shown above the `Working directory` panel
- the `directory_config_module` was removed; instead, `program_config_module` directly calls the shinyDirButton and shinyFilesButton one time each. 
- thus, also the testing changed, and `test-program_config_module.R` now mocks the `shinyFiles` functions: following recommendations, to make this easier to test, I wrapped the `shinyFiles` functions and mock those wrappers

>there is a test failing in the CI, but it works locally. will investigate

this test is removed now

File: R/launch_application.R [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+#'
+#' @return A graphical user interface with the app.
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' launch_application()
+#' }
+launch_application <- function() {
+  runApp(appDir = system.file("application", package = "hiaR"))
```
Comment (maltelueken) [inline R/launch_application.R:13]: Maybe add comment that this points to Shiny and not Java app.

File: R/program_config_module.R [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+#' Creates two panels for the user to choose the paths to the DYNAMO executable
+#' and to the working directory.
+#'
+#' @param id The id for reference in the server.
+#'
+#' @returns Shows an error message when the reference data is not NULL and is
+#' does not have the right content.
+#'
+#' @export
+program_config_ui <- function(id) {
```
Comment (maltelueken) [inline R/program_config_module.R:13]: Perhaps rename `id` to `namespace_id` and give a bit more explanation in the docstring?

File: README.Rmd [hunk 3]
```diff
@@ -29,6 +29,13 @@ You can install the development version of hiaR via `devtools`:
 devtools::install_github("rivm-syso/hiaR")
 ```
 
+## Running the app
+
+You can start the app with `launch_application`:
+```r
+launch_application("path/to/reference_data/")
```
Comment (maltelueken) [inline README.Rmd:36]: This doesn't require an input anymore, right?

**Review by maltelueken — APPROVED** [review]
Looks great!

File: R/program_config_module.R (hunk 2 above)
Comment (f-hafner) [reply R/program_config_module.R:13]: I left the argument as `id`; changing it did not affect the app but the tests and I did not want investigate how to fix it (see also here: https://stackoverflow.com/questions/64391273/how-to-test-modules-within-a-golem-framework-shiny-package). I think using the argument name `id` is consistent with conventions in shiny. But, I did update the docstring to make it more precise that it's the namespace id.

**Pull request merged**